Fix express button display in the Cart block

ralucaStan opened the pull request:
This PR fixes a regression introduced in https://github.com/woocommerce/woocommerce-blocks/pull/8601;

The above PR set a grid display for the express payments area for the Checkout block; that component was reused in the Cart block as well leading to visual bugs

This PR sets the grid layout only for the Checkout block

<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->

<!-- Reference any related issues or PRs here -->

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/10528

<!-- Don't forget to update the title with something descriptive. -->
<!-- If you can, add the appropriate labels -->

#### Accessibility

<!-- If you've changed or added any interactions, check off the appropriate items below. You can delete any that don't apply. Use this space to elaborate on anything if needed. -->

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Other Checks

- [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

### Screenshots

<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->

| Before | After |
| ------ | ----- |
|    ![](https://user-images.githubusercontent.com/1628454/259726408-8adbd785-a1b3-461b-b626-b8dd065103f3.png)    |    <img width="625" alt="Screenshot 2023-08-10 at 17 45 54" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/9ea571af-04e3-461a-8c66-0796c1494b31">   |

### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. activate 2 different express payment options: I used Stripe and PayPal ([see instructions](https://woocommerce.com/document/woocommerce-paypal-payments/#woocommerce-blocks)) and make sure they are active
2. add a product to cart
3. visit the Cart block and make sure buttons are stacked
4. go to the Checkout block and make sure buttons are in a grid
5. repeat the testing in the editor as well
6. repeat the steps with different screen sizes
7. Mini cart does not display express payments so it does not need testing

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested (ie: it makes changes to tests, coding standards, docblocks, etc.). -->
* [ ] Should be tested by the development team exclusively <!-- Check this box if this PR should be tested by the development team exclusively (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Performance Impact

<!-- Please document any known performance impact (positive or negative) here. If negative, provide some rationale for why this is an okay tradeoff or how this will be addressed. -->

### Changelog

> Fix Express Payments buttons display in the Cart block

ralucaStan committed:
Fix express button display in the Cart block

When multiple buttons are displayed they don't appear stacked. This PR fixes a regression;

nielslange reviewed (approved):
Thanks for working on this, @ralucaStan. The fix works as expected. 

While reviewing this PR, I did notice the following things:
- The labels [◼️ block: cart](https://github.com/woocommerce/woocommerce-blocks/labels/%E2%97%BC%EF%B8%8F%20block%3A%20cart) and [◼️ block: checkout](https://github.com/woocommerce/woocommerce-blocks/labels/%E2%97%BC%EF%B8%8F%20block%3A%20checkout) are missing
- "WooCommerce Visibility" hasn't been defined
- Either the changelog message or the label [skip-changelog](https://github.com/woocommerce/woocommerce-blocks/labels/skip-changelog) is missing
- One E2E test is currently failing

That said, as the fix itself is working as expected, I'm preapproving this PR.

ralucaStan committed:
Fix editor experience

ralucaStan commented:
@nielslange thank you for checking the PR description 👏  I updated that. 


<img width="348" alt="Screenshot 2023-08-11 at 16 56 30" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/e995101e-d80c-4372-94ef-8279fd6630ad">
Could you also retest the buttons in the editor? I found a bug in the editor that I just fixed in a second commit. 

Aparently moving the class decreased the specificity and the ul tag received some editor padding

nielslange commented:
> Could you also retest the buttons in the editor? I found a bug in the editor that I just fixed in a second commit.
> Aparently moving the class decreased the specificity and the ul tag received some editor padding

Can you tell me with which theme the padding issue appeared? I'm asking as I didn't see the padding issue during my initial review.

ralucaStan commented:
I was using Storefront, without Gutenberg

nielslange commented:
> I was using Storefront, without Gutenberg

While the PR works as expected when using the Storefront theme, when using the Twenty Twenty-Three theme, the Express Checkout inner block shows an error, both when Gutenberg is activated and deactivated.

<img width="1352" alt="Screenshot 2023-08-14 at 11 03 45" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/fd5953d9-09f8-468a-bd0f-09bb02401cfb">

The console shows the following error message:
```
TypeError: Cannot read properties of undefined (reading 'driver')
    at O (checkout-block.js:242:48)
    at wt (react-dom.min.js?ver=18.2.0:10:47637)
    at vr (react-dom.min.js?ver=18.2.0:10:56452)
    at js (react-dom.min.js?ver=18.2.0:10:121453)
    at wl (react-dom.min.js?ver=18.2.0:10:88659)
    at bl (react-dom.min.js?ver=18.2.0:10:88587)
    at yl (react-dom.min.js?ver=18.2.0:10:88450)
    at il (react-dom.min.js?ver=18.2.0:10:85274)
    at ol (react-dom.min.js?ver=18.2.0:10:83831)
    at v (react.min.js?ver=18.2.0:10:3385)
```

ralucaStan commented:
@nielslange This PR only introduces CSS changes. Can you replicate this issue with `trunk` as well?

ralucaStan commented:
As I've mentioned outside of this PR I can't replicate the issue 

<img width="1691" alt="Screenshot 2023-08-14 at 12 32 20" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/3f105a86-64df-4f09-814c-8277c7d10a11">


<img width="1724" alt="Screenshot 2023-08-14 at 12 31 12" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/8fa4c927-608b-4024-a816-b4a42d0f4877">

If you can't replicate the error on trunk as well then it means the issue is coming from this branch and I would need detailed replication steps to be able to understand your setup better.

opr reviewed (approved):
Looks good! Minor nitpick on the comment on line 8 is that it should start with a capital letter and end with a full stop, sorry 😬
